File: insertion_sort/python/insertion_sort.py
import time
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '../../')))
from list_generator import list_generator

def main():
    unsorted_list = list_generator()

    start = time.time()
    sorted_list = insertion_sort(unsorted_list)
    end = time.time()

    elapsed_time = end - start

    print(f"Time taken by insertion sort: {elapsed_time:.7f} seconds")

def insertion_sort(list):
    for i in range(1, len(list)):
        j = i -1
        temp = list[i]

        while temp < list[j] and j>=0:
            list[j+1] = list[j]
            j -= 1
        
        list[j+1] = temp
    
    return list

if __name__ == "__main__":
    main()

File: insertion_sort/go/insertion_sort.go
package main

import (
	"fmt"
	"time"

	utils "github.com/ggwellplayed4101/Algorithms/list_generator"
)

const size int = 1000

func main() {
	unsorted_list := utils.Random_Arr(size)
	start := time.Now()
	insertion_sort(unsorted_list)
	elapsed := time.Since(start).Seconds()
	fmt.Println("Time taken by insertion sort:", elapsed, "seconds")
}

func insertion_sort(list []int) []int {
	length := len(list)
	for i := 1; i < length; i++ {
		j := i - 1
		temp := list[i]
		for j >= 0 && temp < list[j] {
			list[j+1] = list[j]
			j -= 1
		}
		list[j+1] = temp
	}
	return list
}
